exec/java-exec/src/main/java/org/apache/drill/exec/store/log/LogRecordReader.java

```diff
@@ -0,0 +1,261 @@
+  private static final int MAX_RECORDS_PER_BATCH = 8096;
+
+  private String inputPath;
+  private BufferedReader reader;
+  private DrillBuf buffer;
+  private VectorContainerWriter writer;
+  private LogFormatPlugin.LogFormatConfig config;
+  private int lineCount;
+  private Pattern r;
+
+  private List<String> fieldNames;
+  private List<String> dataTypes;
+  private boolean errorOnMismatch;
+  private String dateFormat;
+  private String timeFormat;
+  private java.text.DateFormat df;
+  private java.text.DateFormat tf;
+  private long time;
+
+  public LogRecordReader(FragmentContext fragmentContext, String inputPath, DrillFileSystem fileSystem,
+                         List<SchemaPath> columns, LogFormatPlugin.LogFormatConfig config) throws OutOfMemoryException {
+    try {
+      Path hdfsPath = new Path(inputPath);
+      Configuration conf = new Configuration();
+      FSDataInputStream fsStream = fileSystem.open(hdfsPath);
+      CompressionCodecFactory factory = new CompressionCodecFactory(conf);
+      CompressionCodec codec = factory.getCodec(hdfsPath);
+      if (codec == null) {
+        reader = new BufferedReader(new InputStreamReader(fsStream.getWrappedStream(), "UTF-8"));
+      } else {
+        CompressionInputStream comInputStream = codec.createInputStream(fsStream.getWrappedStream());
+        reader = new BufferedReader(new InputStreamReader(comInputStream));
+      }
+      this.inputPath = inputPath;
+      this.lineCount = 0;
+      this.config = config;
+      this.buffer = fragmentContext.getManagedBuffer(4096);
+      setColumns(columns);
+
+    } catch (IOException e) {
+      logger.debug("Log Reader Plugin: " + e.getMessage());
+    }
+  }
+
+  public void setup(final OperatorContext context, final OutputMutator output) throws ExecutionSetupException {
+    this.writer = new VectorContainerWriter(output);
+    String regex = config.getPattern();
+
+
+    fieldNames = config.getFieldNames();
+    dataTypes = config.getDataTypes();
+    dateFormat = config.getDateFormat();
+    timeFormat = config.getTimeFormat();
+    errorOnMismatch = config.getErrorOnMismatch();
+
+    /*
+    This section will check for;
+    1.  Empty regex
+    2.  Invalid Regex
+    3.  Empty date string if the date format is used
+    4.  No capturing groups in the regex
+    5.  Incorrect number of data types
+    6.  Invalid data types
+     */
+    if (regex.isEmpty()) {
+      throw UserException.parseError().message("Log parser requires a valid, non-empty regex in the plugin configuration").build(logger);
+    } else {
+      //TODO Check for invalid regex
+      r = Pattern.compile(regex);
+      Matcher m = r.matcher("test");
+      if (m.groupCount() == 0) {
+        throw UserException.parseError().message("Invalid Regular Expression: No Capturing Groups", 0).build(logger);
+      } else if (m.groupCount() != (fieldNames.size())) {
```

Comment:
Suggestion: `!=` -> `>=`. This allows debugging. Define all fields. Then, work out the pattern for the first. Then do the second. Very hard to get all fields right the first time. Nuisance to remove fields temporarily.